pkg/csi/service/wcpguest/controller.go

```diff
@@ -521,75 +521,71 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 func controllerPublishForBlockVolume(ctx context.Context, req *csi.ControllerPublishVolumeRequest, c *controller) (
 	*csi.ControllerPublishVolumeResponse, string, error) {
 	log := logger.GetLogger(ctx)
+	var isVolumePresentInSpec, isVolumeAttached bool
+	var diskUUID string
+	var err error
+
 	virtualMachine := &vmoperatortypes.VirtualMachine{}
 	vmKey := types.NamespacedName{
 		Namespace: c.supervisorNamespace,
 		Name:      req.NodeId,
 	}
 
-	var err error
-	if err = c.vmOperatorClient.Get(ctx, vmKey, virtualMachine); err != nil {
-		msg := fmt.Sprintf("failed to get VirtualMachines for the node: %q. Error: %+v", req.NodeId, err)
-		log.Error(msg)
-		return nil, csifault.CSIInternalFault, status.Errorf(codes.Internal, msg)
-	}
-	// Check if volume is already present in the virtualMachine.Spec.Volumes
-	var isVolumePresentInSpec, isVolumeAttached bool
-	var diskUUID string
-	for _, volume := range virtualMachine.Spec.Volumes {
-		if volume.PersistentVolumeClaim != nil && volume.Name == req.VolumeId {
-			log.Infof("Volume %q is already present in the virtualMachine.Spec.Volumes", volume.Name)
-			isVolumePresentInSpec = true
-			break
-		}
-	}
 	timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
-	// if volume is present in the virtualMachine.Spec.Volumes check if volume's status is attached and DiskUuid is set
-	if isVolumePresentInSpec {
-		for _, volume := range virtualMachine.Status.Volumes {
-			if volume.Name == req.VolumeId && volume.Attached && volume.DiskUuid != "" {
-				diskUUID = volume.DiskUuid
-				isVolumeAttached = true
-				log.Infof("Volume %q is already attached in the virtualMachine.Spec.Volumes. Disk UUID: %q",
-					volume.Name, volume.DiskUuid)
+	timeout := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
+	for {
+		if err = c.vmOperatorClient.Get(ctx, vmKey, virtualMachine); err != nil {
+			msg := fmt.Sprintf("failed to get VirtualMachines for the node: %q. Error: %+v", req.NodeId, err)
+			log.Error(msg)
+			return nil, csifault.CSIInternalFault, status.Errorf(codes.Internal, msg)
+		}
+		// Check if volume is already present in the virtualMachine.Spec.Volumes
+		for _, volume := range virtualMachine.Spec.Volumes {
+			if volume.PersistentVolumeClaim != nil && volume.Name == req.VolumeId {
+				log.Infof("Volume %q is already present in the virtualMachine.Spec.Volumes", volume.Name)
+				isVolumePresentInSpec = true
 				break
 			}
 		}
-	} else {
-		timeout := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
-		for {
-			// Volume is not present in the virtualMachine.Spec.Volumes, so adding
-			// volume in the spec and patching virtualMachine instance.
-			vmvolumes := vmoperatortypes.VirtualMachineVolume{
-				Name: req.VolumeId,
-				PersistentVolumeClaim: &vmoperatortypes.PersistentVolumeClaimVolumeSource{
-					PersistentVolumeClaimVolumeSource: corev1.PersistentVolumeClaimVolumeSource{
-						ClaimName: req.VolumeId,
-					},
+		if isVolumePresentInSpec {
+			break
```

Comment:
This is removing the Volume status check before we break here
`// if volume is present in the virtualMachine.Spec.Volumes check if volume's status is attached and DiskUuid is set` 

Can you run pipeline and make sure all tests are passing

Reply:
the pipeline is breaking since Friday, so am doing manual validation now.
any specific test you want to me execute before I can merge this PR?
I will create PVC and Pod is GC to validate this change.